src/Pact/Repl/Lib.hs

```diff
@@ -709,15 +706,15 @@ envHash i as = argsError i as
 
 setGasLimit :: RNativeFun LibState
 setGasLimit _ [TLitInteger l] = do
-  setenv (eeGasEnv . geGasLimit) (fromIntegral l)
+  setenv (eeGasEnv . geGasLimit) (gasLimitToMicroGasLimit (fromIntegral l))
   return $ tStr $ "Set gas limit to " <> tShow l
 setGasLimit i as = argsError i as
 
 envGas :: RNativeFun LibState
 envGas _ [] = do
-  getGas >>= \g -> return (tLit $ LInteger $ fromIntegral g)
+  getGas >>= \g -> return (tLit $ LInteger $ fromIntegral (microGasToGas g))
```

Comment:
This is a topic for discussion: should we make `env-microgas`? 

Changing `env-gas` potentially breaks so many tests downstream, and it would be easy to make a native that works on the more precise units.